Libraries/DirectMLX.h

```diff
@@ -2079,15 +2080,17 @@ namespace dml
         assert(strides.size() == startPadding.size());
         assert(strides.size() == endPadding.size());
 
-        // Calculate output size
-        TensorDimensions outputSizes;
-        outputSizes.push_back(inputTensor.sizes[0]); // N
-        outputSizes.push_back(inputTensor.sizes[1]); // C
-        for (size_t i = 0; i < windowSizes.size(); ++i)
+        // Calculate output size, if not explicitly provided
```

Comment:
I have another suggestion: DirectML native API requires the output tensor shape which contains the outputSizes, it will not use roundingType as parameter since outputSizes can already predict the roundingType.
While using DirectMlX.h, need you consider the specified roundingType when you calculate the output size if not explicitly provided? Otherwise, the pool2d will always perform as specifying floor roundingType when using the calculated output size. Thanks!

Reply:
Hi, yes, DirectMLX will accept any outputSize here if provided. If an explicit outputSize is not provided, DirectMLX will calculate an appropriately-sized output as a default for you. But if you have a specific requirement for the output tensor size to be smaller or larger than the default, you should compute the outputSize manually and supply it to dml::AveragePool() yourself. For most callers the default outputSizes computed by DirectMLX will be sufficient, but if you have a specific or custom parameter (e.g. a roundingType attribute) that needs to affect the output size, this change will give the flexibility to supply a custom output size.

Reply:
> Hi, yes, DirectMLX will accept any outputSize here if provided. If an explicit outputSize is not provided, DirectMLX will calculate an appropriately-sized output as a default for you. But if you have a specific requirement for the output tensor size to be smaller or larger than the default, you should compute the outputSize manually and supply it to dml::AveragePool() yourself. For most callers the default outputSizes computed by DirectMLX will be sufficient, but if you have a specific or custom parameter (e.g. a roundingType attribute) that needs to affect the output size, this change will give the flexibility to supply a custom output size.

OK, thanks for your explanation, got it. @adtsai